push/sender/src/main/java/org/jboss/aerogear/unifiedpush/message/kafka/InstallationMetricsConsumerRegistrationHook.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * JBoss, Home of Professional Open Source
+ * Copyright Red Hat, Inc., and individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * 	http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.aerogear.unifiedpush.message.kafka;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import javax.ejb.Singleton;
+import javax.ejb.Startup;
+import javax.inject.Inject;
+
+import org.jboss.aerogear.unifiedpush.service.metrics.PushMessageMetricsService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A singleton instance initialized during application start up. It initializes and starts new
+ * {@link InstallationMetricsKafkaConsumer}. When processed of being removed by the container, the customer instance will be
+ * stopped and destroyed.
+ *
+ */
+@Singleton
+@Startup
+public class InstallationMetricsConsumerRegistrationHook {
+
+    private final static Logger logger = LoggerFactory.getLogger(InstallationMetricsConsumerRegistrationHook.class);
+
+    @Inject
+    private PushMessageMetricsService metricsService;
+
+    private InstallationMetricsKafkaConsumer consumer;
+    private ExecutorService executor;
+
+    /**
+     * Start a new instance of {@link InstallationMetricsKafkaConsumer}.
+     */
+    @PostConstruct
+    private void startup() {
+        executor = Executors.newSingleThreadExecutor();
```

Comment:
let's use JEE's managed executerservice here. lookup via JNDI. only create one when there is no managed one available